### ISSUE TITLE
Critical Path CSS

### ISSUE THREAD

**Issue by bdolor** [BCcampus/eypd#276]
is the minimum CSS required to render above the fold content. 
ref:
-
 https://developers.google.com/speed/docs/insights/PrioritizeVisibleContent?hl=
- https://github.com/addyosmani/critical-path-css-tools
- https://addyosmani.com/blog/detecting-critical-above-the-fold-css-with-paul-kinlan-video/?utm_source=wordpress&utm_medium=plugin&utm_term=optimization&utm_campaign=Above%20The%20Fold%20Optimization

**Comment by bdolor** [comment]
see https://github.com/GoogleChromeLabs/ProgressiveWordPress/blob/master/src/theme/header.php#L39 for an implementation strategy in as much as the separated critical css from lazy. Should be fairly straight forward for us given the modular approach we've taken with scss files. Difference of course is that we're a child theme and we don't have control over splitting and loading the parent theme css. So, lowering expectations on how much influence we have over this improving our optimization metric is likely warranted. 
Also interesting to note `<noscript>` tags that include all lazy styles if no javascript is detected.

**Comment by alexpflores** [comment]
I was able to split up the critical css from our main style sheet via #291. No observable performance benefits gained at this scale. 

The same would need to be done to the rest of the css files, along with determining what is critical for the specific page being loaded. There are some WP plugins that help identify your critical css, and get you halfway there. A better approach might be to enable this at the server level. I'm looking into this now. 
 
Google's [mod_PageSpeed](https://www.modpagespeed.com/) is an Apache module for rewriting web pages to reduce latency and bandwidth, it has a filter called [prioritize_critical_css](https://www.modpagespeed.com/doc/filter-prioritize-critical-css) which identify's the CSS rules needed to render the page, and then it `"Parses the CSS and replaces it with inlined CSS that contains only the rules used on the page."` This avoids any blocking CSS requests needed for the initial render.

**Comment by alexpflores** [comment]
Look at it from the perspective of WordPress PWA, figure out how to implement this specifically in WordPress.

**Comment by alexpflores** [comment]
Most critical CSS solutions I've found aren't specific to WordPress and the one I found, rely's on a combination of external paid service and 3-4 plugins. This one will require significantly more time to find an adequate solution. 

I've found A [Laravel package](https://packagist.org/packages/krisawzm/critical-css) for generating and using inline critical-path CSS, but it only gets us halfway there, as there's no integration with WordPress. It will only generate a unique file for each of the URIs (routes) provided.
